mm2src/coins/eth.rs

```diff
@@ -1017,17 +1027,23 @@ impl SwapOps for EthCoin {
     async fn search_for_swap_tx_spend_my(
         &self,
         input: SearchForSwapTxSpendInput<'_>,
-    ) -> Result<Option<FoundSwapTxSpend>, String> {
-        let swap_contract_address = try_s!(input.swap_contract_address.try_to_address());
+    ) -> Result<Option<FoundSwapTxSpend>, MmError<FoundSwapTxSpendErr>> {
+        let swap_contract_address = input
+            .swap_contract_address
+            .try_to_address()
+            .map_to_mm(FoundSwapTxSpendErr::SwapContractAddrError)?;
```

Comment:
`SwapContractAddrError` is not a common error variant. For example, there is no such `swap_contract_address` for UTXO coins.